subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/capabilities/CapabilitiesRulesIntegrationTest.groovy

```diff
@@ -66,7 +66,7 @@ class CapabilitiesRulesIntegrationTest extends AbstractModuleDependencyResolveTe
         failure.assertHasCause("Cannot choose between cglib:cglib-nodep:3.2.5 and cglib:cglib:3.2.5 because they provide the same capability: cglib:cglib:3.2.5")
     }
 
-    def "can detect conflict with capability in different versions"() {
+    def "can detect conflict with capability in different versions and upgrade automatically to latest version"() {
```

Comment:
We need to think about users that specify `ResolutionStrategy.failOnVersionConflict()`. I think we should probably honour this setting when resolving capability versions as well. Do you agree?
If so, let's open a separate issue / PR for that change.

Reply:
I agree, this makes sense to me: it's a conflict. It would however slightly be off the name of the method, which states "on version conflict", because it's on capability conflict, and that there are 2 kinds of conflicts: capability version conflict and capability provider conflict.